### PR TITLE
[[ Bug 14536 ]] Clearing the UQL state of a variable should happen with ...

### PR DESCRIPTION
**Pull request by livecodesebastien**
...the reference parameters as well.

### CODE REVIEW

File: engine/src/variable.cpp [hunk 1]
```diff
@@ -1770,6 +1770,10 @@ void MCVarref::clearuql()
 		ref->clearuql();
 	else if (!isparam)
 		handler->getvar(index, isparam)->clearuql();
+    // SN-2015-02-25: [[ Bug 14536 ]] Reinstate the UQL unsetting
+    //  for the container as well.
+    else
+        handler->getcontainer(index, isparam)->getvar()->clearuql();
```
Comment (runrevmark) [inline engine/src/variable.cpp:1776]: Does handler -> getvar() not return the referenced variable for an @ parameter, if so then this change is the same as removing the (!isparam) guard.

**Comment by runrevmark** [comment]
Having played with this in 6.6.2 then there is a bit of subtlety here which we need to make sure we've captured. It seems that referenced variables only become non-UQLs in the caller if the called handler actually targets them as variables.

In Steve's example, a 'split' is being called on the variable as a reference so that's fine. However, in 6.6.2 the following results in tVar _not_ losing its uql-ness:

```
on mouseUp
  answer testRef(tVar)
end mouseUp

function testRef @xVar
  return xVar
end testRef
```

Can you make sure that the change you make to 7.0 preserves the above?

**Comment by livecodesebastien** [comment]
Yes it preserves it - MCVarref::clearuql() is only called from `MCEngineExecPutIntoVariable`.
As for the `!isparam` guard, I believe that it is here because `handler->getvar(index, isparam)` returns `nil`, if `isparam` is true. It was always unsetting the uql flag in 6.7, since there was no containers.

**Comment by runrevmark** [comment]
Great - thanks for checking - the UQL behavior is quite subtle!

Reviewed-by: @runrevmark.

**Pull request merged**